diff --git a/configure.ac b/configure.ac
index 2a15d72..4a906f1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -262,6 +262,52 @@ AS_CASE([$host_os],
     abi_vis=no
     abi_wcsl=no
   ],
+  [openedition], [
+    api_time_macros=no
+    abi_accmode=no
+    abi_arc4random=no
+    abi_arc4random_stir=no
+    abi_asprintf=no
+    abi_bsd_getopt=no
+    abi_closefrom=yes
+    abi_err=no
+    abi_errc=no
+    abi_expand_number=yes
+    abi_explicit_bzero=yes
+    abi_fgetln=no
+    abi_flopen=yes
+    abi_fmtcheck=no
+    abi_fpurge=no
+    abi_freezero=yes
+    abi_funopen=no
+    abi_getbsize=yes
+    abi_getpeereid=no
+    abi_humanize_number=yes
+    abi_id_from_name=yes
+    abi_inet_net_pton=no
+    abi_md5=yes
+    abi_name_from_id=no
+    abi_nlist=no
+    abi_pidfile=yes
+    abi_proctitle=yes
+    abi_progname=no
+    abi_readpassphrase=no
+    abi_reallocarray=yes
+    abi_reallocf=no
+    abi_recallocarray=yes
+    abi_stringlist=no
+    abi_sort=no
+    abi_strl=no
+    abi_strmode=no
+    abi_strnstr=no
+    abi_strtonum=no
+    abi_strtox=yes
+    abi_timeconv=no
+    # On libmd.
+    abi_transparent_libmd=no
+    abi_vis=no
+    abi_wcsl=no
+  ],
   [solaris2*], [
     api_time_macros=yes
 
