diff --git a/src/flopen.c b/src/flopen.c
index 679445f..027f2e6 100644
--- a/src/flopen.c
+++ b/src/flopen.c
@@ -37,6 +37,14 @@
 #include <unistd.h>
 
 #include <libutil.h>
+#ifdef __MVS__
+#define AT_FDCWD -100
+#define LOCK_SH 1       /* Shared lock.  */
+#define LOCK_EX 2       /* Exclusive lock.  */
+#define LOCK_NB         4       /* or'd with one of the above to prevent
+                                   blocking */
+#define LOCK_UN 8       /* Unlock.  */
+#endif
 
 static int
 lock_file(int fd, int flags)
